tests/integration_test.rs

```diff
@@ -101,84 +151,86 @@ async fn end_to_end_test() {
-        () = otel_long_running_task => {
+        _ = otel_long_running_task => {
             panic!("Otel component ended unexpectedly");
         },
         () = run_tests_task => {
+
         }
     }
 
     // TODO: troubleshoot why calling `otel_component.shutdown()` blocks test execution here.
 
     filtered_target.shutdown_tx.send(()).await.unwrap();
     unfiltered_target.shutdown_tx.send(()).await.unwrap();
+    let () = self_signed_cert.cleanup();
 }
 
+#[allow(clippy::too_many_arguments)]
 async fn run_tests(
-    mut filtered_metrics_rx: Receiver<ExportMetricsServiceRequest>,
-    mut filtered_logs_rx: Receiver<ExportLogsServiceRequest>,
-    mut unfiltered_metrics_rx: Receiver<ExportMetricsServiceRequest>,
-    mut unfiltered_logs_rx: Receiver<ExportLogsServiceRequest>,
+    filtered_metrics_rx: Receiver<ExportMetricsServiceRequest>,
+    filtered_metrics_with_tls_rx: Receiver<ExportMetricsServiceRequest>,
+    filtered_logs_rx: Receiver<ExportLogsServiceRequest>,
+    filtered_logs_with_tls_rx: Receiver<ExportLogsServiceRequest>,
+
+    unfiltered_metrics_rx: Receiver<ExportMetricsServiceRequest>,
+    unfiltered_metrics_with_tls_rx: Receiver<ExportMetricsServiceRequest>,
+    unfiltered_logs_rx: Receiver<ExportLogsServiceRequest>,
+    unfiltered_logs_with_tls_rx: Receiver<ExportLogsServiceRequest>,
     sample_attribute: &Attribute,
+    prom_port: u16,
 ) {
     let meter = global::meter_provider().meter("end_to_end_test");
     let test_counter = meter.u64_counter("test_counter").init();
     test_counter.add(1, &[]);
 
-    // validate that the metric is exported to the OTLP target
-    let metrics_export_request = timeout(Duration::from_secs(2), filtered_metrics_rx.recv())
-        .await
-        .unwrap()
-        .unwrap();
-    let (name, value, temporality) = get_counter(&metrics_export_request);
-    assert_eq!(name, "test_counter");
-    assert_eq!(value, 1);
-    assert_eq!(AggregationTemporality::Cumulative, temporality);
-
-    let kv = KeyValue {
-        key: sample_attribute.key.clone(),
-        value: Some(AnyValue {
-            value: Some(StringValue(sample_attribute.value.clone())),
-        }),
-    };
-    //validate resource attribute
-    assert!(get_resource_attributes(&metrics_export_request).contains(&kv));
-
-    // validate that the metric is exported to the unfiltered OTLP target
-    let metrics_export_request = timeout(Duration::from_secs(2), unfiltered_metrics_rx.recv())
-        .await
-        .unwrap()
-        .unwrap();
-    let (name, value, temporality) = get_counter(&metrics_export_request);
-    assert_eq!(name, "test_counter");
-    assert_eq!(value, 1);
-    assert_eq!(AggregationTemporality::Delta, temporality);
-    assert!(get_resource_attributes(&metrics_export_request).contains(&kv));
+    // validate that the metric is exported to the OTLP targets
+    validate_test_counter(
+        filtered_metrics_rx,
+        sample_attribute,
+        AggregationTemporality::Cumulative,
```

Comment:
Just a question, why does it check if the counter is Delta, not Cumulative as it should be converted in src/lib.rs?

Reply:
We are testing both the Delta and Cumulative paths and the exports in the config have been configured for both of these.